pkg/connection/store/config.go

```diff
@@ -0,0 +1,91 @@
+/*
+ Copyright 2022 The Crossplane Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package store
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	v1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
+)
+
+// An ConfigOption modifies an unstructured store config resource.
+type ConfigOption func(*UnstructuredConfig)
+
+// ConfigWithGroupVersionKind sets the GroupVersionKind of the unstructured
+// store config resource.
+func ConfigWithGroupVersionKind(gvk schema.GroupVersionKind) ConfigOption {
+	return func(c *UnstructuredConfig) {
+		c.SetGroupVersionKind(gvk)
+	}
+}
+
+// ConfigWithConditions returns an Option that sets the supplied conditions
+// on an unstructured store config resource.
+func ConfigWithConditions(c ...v1.Condition) ConfigOption {
+	return func(cr *UnstructuredConfig) {
+		cr.SetConditions(c...)
+	}
+}
+
+// NewConfig returns a new unstructured store config resource.
+func NewConfig(opts ...ConfigOption) *UnstructuredConfig {
+	c := &UnstructuredConfig{unstructured.Unstructured{Object: make(map[string]interface{})}}
+	for _, f := range opts {
+		f(c)
+	}
+	return c
+}
+
+// An UnstructuredConfig is an unstructured store config resource.
+type UnstructuredConfig struct {
+	unstructured.Unstructured
+}
```

Comment:
Per my previous comment, given that this package will be used by each provider on its own `StoreConfig` type, could we avoid using `*Unstructured` and instead define `StoreConfig` structs for each provider? We could generate method sets for them using angry jet, like we do for `ProviderConfig` types.

Reply:
Removed `UnstructuredConfig` type as suggested.